components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -2546,13 +2553,14 @@ subroutine cldliq_immersion_freezing(t,lamc,mu_c,cdist1,qc_incld,    &
    real(rtype), intent(out) :: qcheti
    real(rtype), intent(out) :: ncheti
 
-   real(rtype) :: dum, Q_nuc, N_nuc
+   real(rtype) :: dum1, dum2, Q_nuc, N_nuc
 
    if (qc_incld.ge.qsmall .and. t.le.rainfrze) then
       ! for future: calculate gamma(mu_c+4) in one place since its used multiple times  !AaronDonahue, TODO
-      dum   = (1._rtype/lamc)**3
-      Q_nuc = cons6*cdist1*gamma(7._rtype+mu_c)*exp(aimm*(zerodegc-t))*dum**2
-      N_nuc = cons5*cdist1*gamma(mu_c+4._rtype)*exp(aimm*(zerodegc-t))*dum
+      dum1 = bfb_exp(aimm*(zerodegc-t))
+      dum2 = bfb_cube(1._rtype/lamc)
+      Q_nuc = cons6*cdist1*bfb_gamma(7._rtype+mu_c)*dum1*dum2*dum2
```

Comment:
Here I think you mean to use bfb_square(dum2).

Reply:
Good point.